portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -52,20 +107,41 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
     response.setContentType("application/json");
 
-    Query query = new Query("Comment");
+    String numComments = request.getParameter("numComments");
+    String sortType = request.getParameter("sortType");
+    int nComments = 10;
+
+    if (!numComments.equals("All")) {
+      try {
+        nComments = Integer.parseInt(numComments);
+      } catch (Exception e) {
+      }
+    }
+
+    Query query = new Query("Comment").addSort(
+        "iso8601", sortType.equals("asc") ? Query.SortDirection.ASCENDING
+                                          : Query.SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery storedComments = datastore.prepare(query);
 
     ArrayList<Comment> comments = new ArrayList<Comment>();
+    int maxComments = 0;
     for (Entity entity : storedComments.asIterable()) {
-      String name = (String) entity.getProperty("name");
-      String comment = (String) entity.getProperty("comment");
+      String name = (String)entity.getProperty("name");
+      String comment = (String)entity.getProperty("comment");
+      String date = (String)entity.getProperty("date");
+      String time = (String)entity.getProperty("time");
 
-      comments.add(new Comment(name, comment));
+      maxComments++;
+      comments.add(new Comment(name, comment, date, time));
+      if (!numComments.equals("All") && maxComments >= nComments) {
```

Comment:
It's usually not a good idea to have a string literal appear in multiple places, e.g., if you want to change it later, you might miss some. So you could make a constant for "All". Alternatively, you could set nComments to something really large the first time you checked against "All".